userspace/libsinsp/fdinfo.cpp

```diff
@@ -423,3 +424,14 @@ void sinsp_fdtable::reset_cache()
 {
 	m_last_accessed_fd = -1;
 }
+
+void sinsp_fdtable::lookup_device(sinsp_fdinfo_t* fdi, uint64_t fd)
+{
+	if(fdi->is_file() && fdi->m_dev == 0 && fdi->m_mount_id != 0)
+	{
+		char procdir[SCAP_MAX_PATH_SIZE];
+		snprintf(procdir, sizeof(procdir), "%s/proc/%ld/", scap_get_host_root(), m_tid);
+		fdi->m_dev = scap_get_device_by_mount_id(m_inspector->m_h, procdir, fdi->m_mount_id);
+		fdi->m_mount_id = 0; // don't try again
```

Comment:
I'm wondering if there's a retryable way that `scap_get_device_by_mount_id` can fail. Reading the code, it looks like it should succeed so long as we provide valid arguments, so maybe not. Would it make sense to only set `m_mount_id` to zero if the return from `scap_get_device_by_mount_id` is nonzero? Or would that just be overengineering?

Reply:
I'd rather have a zero device id than try and fail the mountinfo read on every event. This line is only for the case when scap_get_device_by_mount_id returns zero, otherwise we never enter this block twice (we check that m_dev == 0)